src/main/java/org/sert2521/gamename/Log.kt

```diff
@@ -0,0 +1,34 @@
+package org.sert2521.gamename
+
+import edu.wpi.first.wpilibj.Preferences
+
+/** Checks if a log is loggable with the provided log level from SmartDashboard. */
+private fun isLoggable(level: LogLevel) =
+        level >= when (Preferences.getInstance().getString("log_level", "info")) {
+            "verbose" -> LogLevel.VERBOSE
+            "debug" -> LogLevel.DEBUG
+            "warn" -> LogLevel.WARN
+            "error" -> LogLevel.ERROR
+            else -> LogLevel.INFO
+        }
+
+enum class LogLevel { VERBOSE, DEBUG, INFO, WARN, ERROR }
+
+private val autoTag: String get() {
+    Thread.currentThread().stackTrace
+            .filter { it.fileName != "LogKt" }
+            .forEach { return it.className }
+    return ""
+}
+
+fun verbose(tag: String? = null, message: Any) = log(LogLevel.VERBOSE, tag, message)
+fun debug(tag: String? = null, message: Any) = log(LogLevel.DEBUG, tag, message)
+fun info(tag: String? = null, message: Any) = log(LogLevel.INFO, tag, message)
+fun warn(tag: String? = null, message: Any) = log(LogLevel.WARN, tag, message)
+fun error(tag: String? = null, message: Any) = log(LogLevel.ERROR, tag, message)
```

Comment:
@andrewda Yeah, that's how it's done on Android but that's stupid. Anko is doing something really interesting where if you implement their interface, it basically does what my magic method does: https://github.com/Kotlin/anko/blob/master/anko/library/static/commons/src/Logging.kt#L23-L43.

https://github.com/Kotlin/anko/blob/master/anko/library/static/commons/src/Logging.kt#L258